File: apps/android/lib/features/station/control/presentation/widgets/sim_notice.dart
import 'package:flutter/material.dart';

/// A standing caution about the simulation, not a transient error.
///
/// Amber rather than the theme's error red: nothing has gone wrong, the panel
/// simply is not a navigation instrument, and colouring it as a fault would
/// teach the crew to ignore real faults.
class SimNotice extends StatelessWidget {
  const SimNotice({super.key, required this.icon, required this.message});
  final IconData icon;
  final String message;

  static Color _amber(Brightness brightness) =>
      brightness == Brightness.dark
          ? const Color(0xFFE3B341)
          : const Color(0xFF9A6400);

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final tone = _amber(theme.brightness);
    return Container(
      padding: const EdgeInsets.fromLTRB(12, 10, 12, 10),
      decoration: BoxDecoration(
        color: tone.withValues(alpha: 0.10),
        borderRadius: BorderRadius.circular(10),
        border: Border(left: BorderSide(color: tone, width: 3)),
      ),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Icon(icon, size: 16, color: tone),
          const SizedBox(width: 8),
          Expanded(
            child: Text(
              message,
              style: theme.textTheme.bodySmall?.copyWith(
                color: theme.colorScheme.onSurfaceVariant,
                height: 1.35,
              ),
            ),
          ),
        ],
      ),
    );
  }
}

File: apps/windows/src/prana_windows/ui/components/sim_notice.py
from __future__ import annotations

from PySide6.QtCore import Qt
from PySide6.QtWidgets import QFrame, QHBoxLayout, QLabel

from prana_windows.ui.icons import GlyphLabel


class SimNotice(QFrame):
    """A standing caution about the simulation, not a transient error.

    Amber rather than the error red, as on the phone
    (`control/presentation/widgets/sim_notice.dart`): nothing has gone wrong, the
    panel simply is not a navigation instrument, and colouring it as a fault
    would teach the crew to ignore real faults.
    """

    def __init__(self, icon: str, parent=None):
        super().__init__(parent)
        self.setObjectName("SimNotice")
        layout = QHBoxLayout(self)
        layout.setContentsMargins(12, 10, 12, 10)
        layout.setSpacing(8)
        layout.addWidget(GlyphLabel(icon, role="warn_icon", size=16), 0, Qt.AlignTop)
        self._text = QLabel()
        self._text.setObjectName("SimNoticeText")
        self._text.setWordWrap(True)
        layout.addWidget(self._text, 1)

    def set_text(self, text: str) -> None:
        self._text.setText(text)

    def text(self) -> str:
        return self._text.text()


__all__ = ["SimNotice"]
